packages/flutter/lib/src/material/page.dart

```diff
@@ -64,9 +124,6 @@ class MaterialPageRoute<T> extends PageRoute<T> {
   Color get barrierColor => null;
 
   @override
-  bool canTransitionFrom(TransitionRoute<dynamic> nextRoute) => false;
```

Comment:
Not a good way. You could ask it to be passed in as an argument to the route, I guess?